### ISSUE TITLE
Exponential Smoothing with `seasonal periods=1` throws error

### ISSUE THREAD

**Issue by mloning** [statsmodels/statsmodels#5682]
#### Describe the bug
When specifying a seasonal component (either `'mul'` or `'add'`), exponential smoothing throws a ValueError when setting `seasonal_periods=1` . 

#### Code Sample, a copy-pastable example if possible

```python
import numpy as np
from statsmodels.tsa.holtwinters import ExponentialSmoothing
y = np.random.lognormal(size=50)
e = ExponentialSmoothing(y, seasonal='mul', seasonal_periods=1)
e.fit()
```

#### Expected Output

No error. Fitted model. 

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

INSTALLED VERSIONS
------------------
Python: 3.7.3.final.0
OS: Darwin 18.5.0 Darwin Kernel Version 18.5.0: Mon Mar 11 20:40:32 PDT 2019; root:xnu-4903.251.3~3/RELEASE_X86_64 x86_64
byteorder: little
LC_ALL: None
LANG: en_GB.UTF-8

Statsmodels
===========

Installed: 0.9.0 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/statsmodels)

Required Dependencies
=====================

cython: 0.29.7 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/Cython)
numpy: 1.15.4 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/numpy)
scipy: 1.2.1 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/scipy)
pandas: 0.24.2 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/pandas)
    dateutil: 2.8.0 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/dateutil)
patsy: 0.5.1 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/patsy)

Optional Dependencies
=====================

matplotlib: 3.0.3 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/matplotlib)
    backend: module://ipykernel.pylab.backend_inline 
cvxopt: Not installed
joblib: 0.13.2 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/joblib)

Developer Tools
================

IPython: 7.4.0 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/IPython)
    jinja2: 2.10 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/jinja2)
sphinx: 2.0.1 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/sphinx)
    pygments: 2.3.1 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages/pygments)
nose: Not installed
pytest: 4.4.2 (/Users/mloning/.conda/envs/sktime/lib/python3.7/site-packages)
virtualenv: Not installed
</details>

**Comment by bashtage** [comment]
If you have a seasonal period of 1 you don't have a seasonal period. What are you trying to accomplish?

**Comment by mloning** [comment]
Yes I know, not sure why it throws an error though rather than simply ignoring it? This came up when running multiple models on multiple datasets with different seasonal periodicity.

**Comment by bashtage** [comment]
This is by design.  It rejects models it knows it cannot estimate.

**Comment by mloning** [comment]
Okay, thanks for the reply! Not sure why it throws an error in `fit` rather than during construction, or even better perhaps simply return the fitted non-seasonal model?

**Comment by bashtage** [comment]
A PR to move it to __int__ would be welcome, and I agree, that is where model spec should be checked. As a rule explicit is better than implicit, and so returning a model different from what was asked for should be avoided since it creates other surprises (like many different parameterizations ultimately being the same model).

**Comment by bashtage** [comment]
This was fixed in 6a6ad5c7937c8b23a6d6720d5a1f3aafd28b7ad6.
